rna/rna-alignment/rna-alignment.wdl

```diff
@@ -0,0 +1,187 @@
+##     falling back to a full-price VM.
+##
+## Outputs:
+## - Per-sample
+##   - <sample-id>.Aligned.sortedByCoord.out.bam
+##   - <sample-id>.SJ.out.tab
+##   - <sample-id>.Chimeric.out.junction
+##   - <sample-id>.Log.final.out
+##   - <sample-id>.Log.out
+##   - <sample-id>.Log.progress.out
+##   - <sample-id>._STARgenome
+##     - sjdbInfo.txt
+##     - sjdbList.out.tab
+##   - <sample-id>._STARpass1
+##     - Log.final.out
+##     - SJ.out.tab
+
+workflow RNAAlignment {
+  String sample_name
+  Array[String] fastq_1
+  Array[String] fastq_2
+
+  # WDL does not support localizing a directory.
+  # Workaround that by passing a TAR file.
+  File star_index
+
+  String star_docker
+  Int star_vm_disk_size_gb
+  String star_vm_memory
+  String runtime_zones
+  Int preemptible_tries
+
+  call star_align {
+    input:
+      fastq_1 = fastq_1,
+      fastq_2 = fastq_2,
+      sample_name = sample_name,
+      star_index = star_index,
+      star_vm_disk_size_gb = star_vm_disk_size_gb,
+      star_vm_memory = star_vm_memory,
+      star_docker = star_docker,
+      runtime_zones = runtime_zones,
+      preemptible_tries = preemptible_tries,
+   }
+
+  output {
+    Array[File] outputFiles = star_align.outputFiles
+    Array[File] outputStarGenomeFiles = star_align.outputStarGenomeFiles
+    Array[File] outputStarpass1Files = star_align.outputStarpass1Files
+  }
+}
+
+task star_align {
+
+  Array[File] fastq_1
+  Array[File] fastq_2
+  String sample_name
+  File star_index
+
+  Int star_vm_disk_size_gb
+  String star_vm_memory
+  String star_docker
+  String runtime_zones
+  Int preemptible_tries
+
+  command<<<
+    set -o errexit
+    set -o nounset
+    set -o pipefail
+    set -o xtrace
+
+    # For each FASTQ1 file, extract fields to build up a list
+    # of readgroup tags.
+    RGTAGLIST=""
+    for FILE in ${sep=" " fastq_1}; do
+      SAMPLE_NAME="$(basename $FILE | cut -d. -f1)"
+
+      LINE="$(head -n 1 < <(zcat $FILE))"
+      FLOWCELL="$(echo $LINE | cut -d: -f3)"
+      LANE="$(echo $LINE | cut -d: -f4)"
+      INDEX="$(echo $LINE | cut -d: -f10)"
+
+      if [[ -z $LINE ]]; then
+        2>&1 "ERROR: Unable to parse first line from $FILE"
+        exit 1
+      fi
+      if [[ -z $FLOWCELL ]]; then
+        2>&1 "ERROR: Unable to parse flowcell from $LINE"
+        exit 1
+      fi
+      if [[ -z $LANE ]]; then
+        2>&1 "ERROR: Unable to parse lane from $LINE"
+        exit 1
+      fi
+      if [[ -z $INDEX ]]; then
+        2>&1 "ERROR: Unable to parse index from $LINE"
+        exit 1
+      fi
+
+      RGTAG="ID:$FLOWCELL.$INDEX.$LANE PL:ILLUMINA SM:$SAMPLE_NAME "
+      RGTAGLIST="$RGTAGLIST , $RGTAG"
+    done
+    RGTAGLIST="$(echo $RGTAGLIST | sed 's/^, //')"
+
+    # Untar the STAR index
+    mkdir star_index
+    tar xfz "${star_index}" --directory star_index --strip-components=1
+
+    # Run STAR --runMode alignReads
+    mkdir output_dir
+    STAR \
```

Comment:
Let's discuss optionally using `timeout` to prevent long-running workflows from wasteful preemption loops.